pytorch_lightning/lite/lite.py

```diff
@@ -233,17 +238,18 @@ def _setup_dataloader(
                 )
             sampler = self._get_distributed_sampler(dataloader, **self._strategy.distributed_sampler_kwargs)
 
-        kwargs = TrainerDataLoadingMixin._get_dataloader_init_kwargs(dataloader, sampler)
-        device = self.device if move_to_device else None
-        if isinstance(self._strategy, TPUSpawnPlugin):
-            dataloader = DataLoader(**kwargs)
-        else:
-            dataloader = _LiteDataLoader(device=device, **kwargs)
-
+        dataloader_kwargs = TrainerDataLoadingMixin._get_dataloader_init_kwargs(dataloader, sampler)
+        try:
+            dataloader = type(dataloader)(**dataloader_kwargs)
+        except TypeError:
+            dataloader_kwargs.pop("dataset")
```

Comment:
why are we re-trying to populate the arguments without dataset?

Reply:
Because a custom dataloader might not take a dataset. Actually it is pretty likely if you subclass it.

Example:

```py
class CustomDataLoader(DataLoader):

    def __init__(self, start, end, *args, **kwargs):
        super().__init__(range(start, end), *args, **kwargs)
```

When doing the re-instantiation, there would be 2 datasets being provided to the super method.

Reply:
There are infinitely many ways one can construct subclasses and call super() with different arguments that don't match the init. So this is an extreme assumption